app/logic/Auth/OAuth2.ts

```diff
@@ -182,16 +184,27 @@ export class OAuth2 extends Observable {
    * @throws OAuth2Error
    */
   protected async getAccessTokenFromParams(params: any, additionalHeaders?: any, tokenURL: string | void = this.tokenURL): Promise<string> {
-    params.scope = this.scope;
     params.client_id = this.clientID;
-    params.client_secret = this.clientSecret || undefined;
+    
+    if (this.clientSecret) {
+      params.client_secret = this.clientSecret;
+      params.scope = this.scope;
+    } else {
+      assert(!!this.codeVerifier, "Need code verifier");
+      params.code_verifier = this.codeVerifier;
+    }
+
+    let headers = {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Accept': 'application/json',
+      ...additionalHeaders,
+    }
+    if (!this.clientSecret) {
+      headers.Authorization = `Basic ${btoa(this.clientID + ":")}`;
```

Comment:
Move `let headers = {` up, then this line can go after `params.code_verifier =`

Reply:
Why do we need an Auth: header when there's no client secret?

This has not been needed so far.

Reply:
I'm not sure. PKCE is mostly just the addition of `code_challange` and `code_challenge_method` when requesting authorization. And the `code_verifier` when getting the token. The changes are mostly a recreation of a successful request on Postman. I'm still not able to track down the root cause.

Reply:
> The changes are mostly a recreation of a successful request on Postman. I'm still not able to track down the root cause.

I see.

1. Check whether it works with all your changes here. I.e. you can successfully log in to Yahoo using OAuth2. If so:
2. Remove changes here 1 by 1, and see whether it still works. If so:
3. Remove the change from the PR.

For those changes that you found are necessary, separate them into 2 PRs for PKCE and other changes, as mentioned, please.